util/src/main/java/org/hyperledger/besu/util/NetworkUtility.java

```diff
@@ -98,4 +104,23 @@ public static void checkPort(final int port, final String portTypeName) {
               "%s port requires a value between 1 and 65535. Got %d.", portTypeName, port));
     }
   }
+
+  public static boolean isPortAvailableForTcp(final int port) {
+    try (final ServerSocket serverSocket = new ServerSocket()) {
+      serverSocket.setReuseAddress(true);
+      serverSocket.bind(new InetSocketAddress(port));
+      return true;
+    } catch (IOException ex) {
+      LOG.trace(String.format("Failed to open port %d for TCP", port), ex);
+    }
+    return false;
+  }
+
+  public static void checkPortsAvailable(final int tcpPort) {
```

Comment:
then method name says `Ports` so suggests that more than one port as parameter like a var arg, otherwise use the singular

Reply:
Changed the name of the method, thanks for raising that @fab-10

Reply:
Changed the approach to cover the ports we will require to initialise all the services defined in the config. So we can have a more general solution that loops through the ports and check if they are available.